yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClientSchedulerBackend.scala

```diff
@@ -64,7 +68,7 @@ private[spark] class YarnClientSchedulerBackend(
     // SPARK-8851: In yarn-client mode, the AM still does the credentials refresh. The driver
     // reads the credentials from HDFS, just like the executors and updates its own credentials
     // cache.
-    if (conf.contains("spark.yarn.credentials.file")) {
+    if (!conf.contains(PRINCIPAL.key) && conf.contains("spark.yarn.credentials.file")) {
```

Comment:
Since you're touching this, you could also use `CREDENTIALS_FILE_PATH`.

BTW I'm not entirely sure why this change is needed, or related to the issue you reported. You want to start the token renewer code exactly when `PRINCIPAL` is defined. You're pretty much disabling this code now, because `CREDENTIALS_FILE_PATH` is never set when `PRINCIPAL` is not.